filebeat/harvester/harvester.go

```diff
@@ -17,12 +17,15 @@ import (
 	"fmt"
 	"regexp"
 
+	"github.com/satori/go.uuid"
+
 	"github.com/elastic/beats/filebeat/config"
 	"github.com/elastic/beats/filebeat/harvester/encoding"
 	"github.com/elastic/beats/filebeat/input"
 )
 
 type Harvester struct {
+	Id                 uuid.UUID
```

Comment:
If the haverster ID is `{{beat.UUID}}-hv{{havester ID}}` then you can find logs and stats associated with a specific harvester, and you can also implicitly associate those logs and stats to the specific beat without having to maintain a mapping of havester ID to owning beat UUID.